File: backend/app/models/baby.py
from datetime import date, datetime

from sqlalchemy import Date, DateTime, ForeignKey, String, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base


class Baby(Base):
    __tablename__ = "babies"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    owner_user_id: Mapped[int] = mapped_column(ForeignKey("users.id", ondelete="CASCADE"), index=True)
    name: Mapped[str] = mapped_column(String(100))
    birth_date: Mapped[date] = mapped_column(Date)
    gender: Mapped[str] = mapped_column(String(10), default="UNKNOWN")  # MALE | FEMALE | UNKNOWN
    profile_image_url: Mapped[str | None] = mapped_column(String(500))
    is_active: Mapped[bool] = mapped_column(default=True)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
    )

    owner = relationship("User", back_populates="babies")

File: frontend/src/features/baby/types/baby.ts
export type BabyGender = "MALE" | "FEMALE" | "UNKNOWN";

export type Baby = {
  id: number;
  name: string;
  birthDate: string;
  gender: BabyGender;
  ageInDays: number;
  isActive: boolean;
};

export type BabyCreateInput = {
  name: string;
  birthDate: string;
  gender: BabyGender;
};

export type BabyUpdateInput = {
  name?: string;
  gender?: BabyGender;
};

